clients/src/main/java/org/apache/kafka/clients/admin/internals/DeleteRecordsHandler.java

```diff
@@ -79,15 +79,15 @@ public static SimpleAdminApiFuture<TopicPartition, DeletedRecords> newFuture(
     @Override
     public DeleteRecordsRequest.Builder buildBatchedRequest(int brokerId, Set<TopicPartition> keys) {
         Map<String, DeleteRecordsRequestData.DeleteRecordsTopic> deletionsForTopic = new HashMap<>();
-        for (Map.Entry<TopicPartition, RecordsToDelete> entry: recordsToDelete.entrySet()) {
-            TopicPartition topicPartition = entry.getKey();
+        for (TopicPartition topicPartition : keys) {
+            RecordsToDelete toDelete = recordsToDelete.get(topicPartition);
             DeleteRecordsRequestData.DeleteRecordsTopic deleteRecords = deletionsForTopic.computeIfAbsent(
                     topicPartition.topic(),
                     key -> new DeleteRecordsRequestData.DeleteRecordsTopic().setName(topicPartition.topic())
             );
             deleteRecords.partitions().add(new DeleteRecordsRequestData.DeleteRecordsPartition()
                     .setPartitionIndex(topicPartition.partition())
-                    .setOffset(entry.getValue().beforeOffset()));
+                    .setOffset(toDelete.beforeOffset()));
```

Comment:
Would it be unduly paranoid to check for `null` before accessing `toDelete`? The call stack up to this point is pretty twisty. I'm assuming that `keys` _shouldn't_ contain any `TopicPartition`s that aren't keys in `recordsToDelete`, but... 🤷‍♂️

Reply:
Yes, in my opinion it would be unduly paranoid. I would end up writing conditional logic for a situation which will never occur, and it would differ from other code in this area of AdminClient which also uses a map which is accessed in this way confidently expecting the entry to be present.